tests/integration-tests/tests/common/utils.py

```diff
@@ -231,3 +235,44 @@ def get_route_tables(subnet_id, region):
         Filters=[{"Name": "association.subnet-id", "Values": [subnet_id]}]
     )
     return [table["RouteTableId"] for table in response["RouteTables"]]
+
+
+def run_system_analyzer(scheduler_commands, remote_command_executor, request):
+    """Run script to collect system information on head and a compute node of a cluster"""
+
+    out_dir = request.config.getoption("output_dir")
+    local_result_dir = f"{out_dir}/system_analyzer"
+    compute_node_shared_dir = "/opt/parallelcluster/shared"
+    head_node_dir = "/tmp"
+
+    logging.info(f"Retrieve head node system information for test: {request.node.name}")
+    result = remote_command_executor.run_remote_script(SYSTEM_ANALYZER_SCRIPT, args=[head_node_dir])
+    logging.debug(f"result.failed={result.failed}")
+    logging.debug(f"result.stdout={result.stdout}")
+    logging.info(
+        f"Copy results from remote cluster into: "
+        f"{local_result_dir}/system_information_head_node_{request.node.name}.tar.gz"
+    )
+    os.makedirs(f"{local_result_dir}", exist_ok=True)
+    remote_command_executor.get_remote_files(
+        f"{head_node_dir}/system-information.tar.gz",
+        f"{local_result_dir}/system_information_head_node_{request.node.name}.tar.gz",
+        preserve_mode=False,
+    )
+    logging.info("Head node system information correctly retrieved.")
+
+    logging.info(f"Retrieve compute node system information for test: {request.node.name}")
+    result = scheduler_commands.submit_script(SYSTEM_ANALYZER_SCRIPT, script_args=[compute_node_shared_dir])
+    job_id = scheduler_commands.assert_job_submitted(result.stdout)
+    scheduler_commands.wait_job_completed(job_id)
+    scheduler_commands.assert_job_succeeded(job_id)
+    logging.info(
```

Comment:
Since head and compute use different method (`remote_command_executor.run_remote_script` and `scheduler_commands.submit_script`) to run the script, the function would be a duplicate of  `run_system_analyzer` or `remote_command_executor.get_remote_files`